src/core/thread/mle.cpp

```diff
@@ -246,6 +252,7 @@ ThreadError Mle::Stop(void)
         mNetif.RemoveUnicastAddress(mLeaderAloc);
     }
 
+    mNetif.GetPendingDataset().StopDelayTimer();
```

Comment:
Do we have all cases covered?  What about the case where we successfully attach but detach before properly receiving the Active/Pending Datasets from the Parent.  Will we restart the Delay Timer in this case?